README.md

```diff
@@ -135,3 +135,127 @@ cargo update -p pallet-ewx-worker-solution
+Extrinsics can be send from `Developer/extrinsic` menu. Extrinsics of Worker pallet are available in `workerNodePallet`. Before sending extrinsic choose one of the development accounts.
+
+### Signing up solution registrar
+
+- choose `signupSolutionRegistrar` extrinsic
+- specify extrinsic parameters:
+  - `friendly name`: `registrar name`,
+  - `legalLocation`: `registrar location`
+- click `Submit Transaction` and in next pop-up click `Sign and Submit`
+- go to Network-Explorer and make sure that there is `workerNodePallet.NewSolutionRegistrarSignup` in the list of recent events
+
+### Registering a solution
+
+- select same account which was used to signup solution registrar
+- choose `registerSolution` extrinsic
+- specify extrinsic parameters:
+  - `namespace`: `solution namespace`
+  - `name`: `solution name`
+  - `description`: `solution description`
+  - `publisherInfo`: `solution publisher info`
+  - `logoUrl`: `solution logo url` // it is optional
+  - `workLogic`: `work logic`
+  - `executionEnvironment`: `10`
+  - `expirationBlock`: `10000`
+  - `maxWaitingThreshhold`: `60`
+  - `voteThresholdPercent`: `60`
+- click `Submit Transaction` and in next pop-up click `Sign and Submit`
+- go to Network-Explorer and make sure that there is
+`workerNodePallet.SolutionCreated` in the list of recent events
+
+## Using Worker pallet from JS/TS
+
+### Installation
+
+The polkadot.js library can be installed using the instructions here: https://polkadot.js.org/docs/api/start/install/ .
+
+Use of Typescript may require additional libraries such as `@polkadot/typegen`. See https://polkadot.js.org/docs/api/start/typescript.user.
+
+### State Queries
+
+Pallet state queries can be done via the `.query` API, as shown [here](https://polkadot.js.org/docs/api/start/api.query)
+
+## Example scenarios
+
+### Solution Registrar Flow
+
+Use the follow script as an example to register solution.
+`ALICE` is one of the predefined development accounts. This script first
+signups ALICE as solution registrar, which allows next to register solution.
+
+```TS
+import { ApiPromise, WsProvider, Keyring } from "@polkadot/api";
+import { blake2AsHex } from "@polkadot/util-crypto";
+
+async function main(): Promise<void> {
+  const wsProvider = new WsProvider("ws://localhost:9947");
+  const api = await ApiPromise.create({ provider: wsProvider });
+
+  const keyring = new Keyring({ type: "sr25519" });
+  const ALICE_KEYRING = keyring.addFromUri("//Alice", {
+    name: "Alice default",
+  });
+  const ALICE_ADDRESS = ALICE_KEYRING.address; // "5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY"
+
+  const registrar_name = "Alice";
+  const registrar_legal_location = "Alice place";
+  await new Promise<void>(async (resolve) => {
```

Comment:
Not sure if this is the best way to ensure that registrar signed up before registering solution

Reply:
I'm not sure either. How does this code work?
```
await new Promise<void>(async (resolve) => {
    let unsub = await api.tx.workerNodePallet
      .signupSolutionRegistrar(registrar_name, registrar_legal_location)
      .signAndSend(ALICE_KEYRING, ({ status }) => {
        if (status.isFinalized) {
          unsub();
          resolve();
        }
      });
  });
```
Is the callback passed to `signAndSend` called multiple times (until `unSub`) is called?

By the way, I think full words are generally better than abbreviations as it is clearer, so I would replace `unSub` with `unSubscribe`

Reply:
Callback indeed called each time when extrinsic status has been changed. 
I'll rename `unsub`

Reply:
From logs I can see that this extrinsic passes Ready -> Broadcast -> InBlock -> Finalized